### PR TITLE
Add test for duplicate data when resaving model to fits

### PR DESCRIPTION
**Pull request by braingram**
When resaving a model to fits, asdf loses track of array-hdu links and saves arrays in both hdu and as internal blocks in the asdf tree. This can make an asdf tree that is larger than what can be stored in a fits hdu.

See asdf issue: https://github.com/asdf-format/asdf/issues/1232

This test is skipped as it fails with the current asdf version.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: jwst/datamodels/tests/test_fits.py [hunk 1]
```diff
@@ -4,7 +4,8 @@
 import numpy as np
 import pytest
 
-from jwst.datamodels import JwstDataModel, RampModel
+import jwst.datamodels
```
Comment (jdavies-st) [inline jwst/datamodels/tests/test_fits.py:7]: As elsewhere in the package, this import is preferred:

```suggestion
from jwst import datamodels
```

File: jwst/datamodels/tests/test_fits.py [hunk 2]
```diff
@@ -68,3 +69,24 @@ def test_casting():
...
+    """
+    fn1 = tmp_path / "test1.fits"
+    fn2 = tmp_path / "test2.fits"
+
+    arr = np.zeros((1000, 100), dtype='f4')
+    m = ImageModel(arr)
+    m.save(fn1)
+
+    m2 = jwst.datamodels.open(fn1)
+    m2.save(fn2)
```
Comment (jdavies-st) [inline jwst/datamodels/tests/test_fits.py:89]: Use the context manager to make sure it is closed on exit.

```suggestion
with datamodels.open(fn1) as m2:
    m2.save(fn2)
```

File: jwst/datamodels/tests/test_fits.py [hunk 3]
```diff
@@ -68,3 +69,24 @@ def test_casting():
...
         dm.data = dm.data + 2
         assert dm.data.sum() > total
+
+
+@pytest.mark.skip(reason='fix required for asdf')
+def test_resave_duplication_bug(tmp_path):
+    """
+    An issue in asdf (https://github.com/asdf-format/asdf/issues/1232)
+    resulted in duplication of data when a model was read from and then
+    written to a fits file.
```
Comment (jdavies-st) [inline jwst/datamodels/tests/test_fits.py:79]: You can mention something about the numpy arrays get written to the `hdu.data` members and remained stored in the asdf tree bundled in the ASDF extension.

**Comment by braingram** [comment]
Thanks! I made the improvements your described. Let me know if there's anything else and if you think this should either be merged now (with the test skipped) or wait until there's a fix for the bug in asdf. Either way I'll mark it ready for review.

**Review by jdavies-st — APPROVED** [review]
Looks great!  Love the proactive test.  👍

**Review by nden — APPROVED** [review]

**Comment by hbushouse** [comment]
Shall we go ahead and merge this now, because it's being skipped and hence can't cause any damage, or wait for fix in asdf to be released (which could be a while)?

**Review by nden — APPROVED** [review]
